scrapers/al/bills.py

```diff
@@ -60,70 +25,132 @@ def scrape(self, session):
             "Referer": "https://alison.legislature.state.al.us/",
         }
 
-        # WARNING: 2023 session id is currently hardcoded
-        json_data = {
-            "query": '{allInstrumentOverviews(instrumentType:"B", instrumentNbr:"", body:"", sessionYear:"2023", sessionType:"2023 Regular Session", assignedCommittee:"", status:"", currentStatus:"", subject:"", instrumentSponsor:"", companionInstrumentNbr:"", effectiveDateCertain:"", effectiveDateOther:"", firstReadSecondBody:"", secondReadSecondBody:"", direction:"ASC"orderBy:"InstrumentNbr"limit:"15"offset:"0"  search:"" customFilters: {}companionReport:"", ){ ID,SessionYear,InstrumentNbr,InstrumentSponsor,SessionType,Body,Subject,ShortTitle,AssignedCommittee,PrefiledDate,FirstRead,CurrentStatus,LastAction,ActSummary,ViewEnacted,CompanionInstrumentNbr,EffectiveDateCertain,EffectiveDateOther,InstrumentType,IntroducedUrl }}',
-            "operationName": "",
-            "variables": [],
-        }
+        offset = 0
+        limit = 10000
+        while True:
```

Comment:
These sorts of loops always make me nervous. Is there any way we can have a "for sure" break?

Reply:
Hmm it'll have to break at 
```
            # no need to paginate again if we max the last page
            if len(page["data"]["allInstrumentOverviews"]) < limit:
                return
```

I could hardcode in a guard at 50k or something, but open to more elegant ideas.